### DISCUSSION TITLE
Dúvida: Query e Conexão e Horse

### DISCUSSION THREAD

**Discussion by AlexhaifaGarcia** [HashLoad/horse#389] [Q&A]
Olá Pessoal, tudo bem?

Eu tenho uma dúvida a respeito de trabalhar com conexão para SQL e requisição horse.

Eu criei um endpoint para meu Get, porém muitos computadores irão consumir esse mesmo endpoint. Tenho um componente no form de TSQLConection e também TSQLQuery no qual uso para me conectar e rodar as minhas requisições. Minha pergunta é:

Essa é a melhor pratica para trabalhar com conexão x query x requisição ?

Terei algum problema com varias pessoas chamando a mesma requisição ao mesmo tempo?

Se tiverem algum video que possam me apontar para estudo, agradeço muito também

Obrigado a quem puder responder.
Alexandre

**Comment by viniciussanchez** [comment]
Bom dia.. sim terá se todos usarem a mesma instância ou tiver trabalhando com variáveis globais

**Comment by AlexhaifaGarcia** [comment]
Bom dia Vinicius

Eu não uso variáveis globais, 

Quando vc fala a mesma instância, poderia ser mais claro, pois o assunto é novo para mim e não ficou claro.

Obrigado

**Reply by AlexhaifaGarcia** [reply]
Vinicius, me tire uma dúvida.
Quando uma API chega no servidor solicitando uma requisição seja ela por get, post, put ou delete, uma thread é criada para essa requisição? Então se eu criar os meus componentes dentro da minha requisição, TSQLconnection e TSQLQuery e apos o seu uso, excluir os componentes, não terei problema com chamadas simultaneas. 
Estou certo a respeito?

Grato

**Reply by viniciussanchez** [reply]
Isso mesmo... cada requisição é uma thread separada

**Comment by AlexhaifaGarcia** [comment]
Muito obrigado Vinicius,
Agora tenho uma direção a seguir.
Uma ótima quarta feira a você
